VulkanHppGenerator.cpp

```diff
@@ -2614,6 +2614,24 @@ void VulkanHppGenerator::appendStruct(std::string & str, std::pair<std::string,
   }
 }
 
+void VulkanHppGenerator::appendStructAssignmentOperator(std::string &str, std::pair<std::string, StructureData> const& structData, std::string const& prefix) const
+{
+  // we need an assignment operator if there is constant sType in this struct
+  if ((nonConstSTypeStructs.find(structData.first) == nonConstSTypeStructs.end()) && !structData.second.members.empty() && (structData.second.members.front().name == "sType"))
+  {
+    assert((2 <= structData.second.members.size()) && (structData.second.members[1].name == "pNext"));
+
+    static const std::string stringTemplate = R"(
+${prefix}${structName} & operator=( ${structName} const & rhs ) VULKAN_HPP_NOEXCEPT
+${prefix}{
+${prefix}  memcpy( &pNext, &rhs.pNext, sizeof( ${structName} ) - offsetof( ${structName}, pNext ) );
```

Comment:
Do you really want to skip `sizeof(void *)` bytes at the end of every struct with the `operator =`? Seems test `DeviceFunctions` crashed not only because `VK_NV_RAY_TRACING_EXTENSION_NAME` is missing among `vk::DeviceCreateInfo::ppEnabledExtensionNames`, but also due to this wrong copy constructor.
I would prefer to save `pNext` then `memcpy` whole `rhs` into `*this` then restore original `pNext` of `*this`.

Reply:
Even better is to generate member-wise copy assignment operator. To be consistent with how arranged copy constructor.

Reply:
This copy assignment operator copies everything but the constant member sType at the beginning of the structure. I don't see how there should be something missing at the end.

Reply:
I wrongly thought, that `pNext` is the first member, sorry.